### PR TITLE
[DX-725] Add extras in the message for channel push notifications

### PR DESCRIPTION
**Pull request by maratal**
It is not support of push notifications with deviceId or clientId. Just have added `extras.push` support for channel pushes. Tried locally with

`./bin/run.js channels publish exampleChannel1 '{"data":"Test","extras":{"push":{"notification":{"title":"Hello","body":"World!"}}}}' --api-key "***"`

and it works.

Jira issue: https://ably.atlassian.net/browse/DX-725

---

> [!NOTE]
> **Low Risk**
> Small, additive change to CLI message shaping plus unit tests; low risk aside from potential edge cases in how JSON input is mapped into `data` vs `extras`.
> 
> **Overview**
> `ably channels publish` now supports publishing messages with `extras` (notably `extras.push` for push notification metadata) when provided in the JSON payload, and avoids duplicating `extras` inside `data`.
> 
> Message preparation also skips setting `data` when the JSON payload contains only `extras`/`name` (i.e., no `data` and nothing else), and the test suite adds coverage for `extras`+`data`, `extras`-only, and `name`+`extras` scenarios.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d3575be52cec1d8d1cb02352d7b8c6bcd2d983d5. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added support for including push notification metadata (extras.push) in Ably channel message publish payloads, enabling push notification data to be transmitted alongside messages.

* **Documentation**
  * Updated product requirements documentation to reflect the new push notification metadata publishing capability.

* **Tests**
  * Added test coverage to verify push notification metadata is correctly passed through the publish flow.

### CODE REVIEW

File: src/commands/channels/publish.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export default class ChannelsPublish extends AblyBaseCommand {
     '$ ably channels publish --transport realtime my-channel "Using realtime transport"',
     '$ ably channels publish my-channel "Hello World" --json',
     '$ ably channels publish my-channel "Hello World" --pretty-json',
+    '$ ably channels publish my-channel \'{"data":"Push notification","extras":{"push":{"notification":{"title":"Hello","body":"World"}}}}\'',
```
[the remark this answers is not in the document]
Comment (maratal) [reply src/commands/channels/publish.ts]: @AndyTWF should we go with this or changes still needed?

File: src/commands/channels/publish.ts [hunk 2]
```diff
@@ -206,6 +207,13 @@ export default class ChannelsPublish extends AblyBaseCommand {
       delete messageData.name;
     }
 
+    // Add extras if provided in the message data (before processing data)
+    if ("extras" in messageData) {
+      message.extras = messageData.extras;
+      // Remove extras from messageData to avoid duplication in data
+      delete messageData.extras;
+    }
```
[the remark this answers is not in the document]
Comment (sacOO7) [reply src/commands/channels/publish.ts]: @coderabbitai wdyt above above comment? Give me detailed explaination for the same

File: src/commands/channels/publish.ts [hunk 3]
```diff
@@ -206,11 +207,18 @@ export default class ChannelsPublish extends AblyBaseCommand {
...
+
     // If data is explicitly provided in the message, use it
     if ("data" in messageData) {
       message.data = messageData.data;
-    } else {
-      // Otherwise use the entire messageData as the data
+    } else if (typeof messageData === "object" && Object.keys(messageData).length > 0 || typeof messageData === "string") {
```
[the remark this answers is not in the document]
Comment (maratal) [reply src/commands/channels/publish.ts]: fixed

File: src/commands/channels/publish.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (maratal) [reply src/commands/channels/publish.ts]: I've unresolved because turned out the code was almost correct and we just mislead each other. Deletion "to avoid duplication in data" exists to not include extracted fields in the `data` in case a user provided object doesn't contain `data` field itself. Now I've only added check for emptiness before assigning to the data. @sacOO7

File: src/commands/channels/publish.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sacOO7) [reply src/commands/channels/publish.ts]: Since, `ably push` command is not mature yet, for now, we have to rely on channel message `extras` field to send notifs. to given channel ( doesn't support to send to particular `deviceId` or `clientId` )
This will be addressed once push notif PR -> https://github.com/ably/ably-cli/pull/126 is merged
